src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -47,23 +48,37 @@ public class ServiceEndpoint
         /// </summary>
         public EndpointMetrics EndpointMetrics { get; internal set; } = new EndpointMetrics();
 
+        private ServiceEndpoint(EndpointType type, string name = "", string version = "1.0")
+        {
+            Name = name;
+            EndpointType = type;
+            Version = version;
+        }
+
         public ServiceEndpoint(string key, string connectionString) : this(connectionString)
         {
             (Name, EndpointType) = ParseKey(key);
         }
 
-        public ServiceEndpoint(string connectionString, EndpointType type = EndpointType.Primary, string name = "")
+        public ServiceEndpoint(TokenCredential credential,
+                               string endpoint,
+                               int? port = 443,
+                               EndpointType type = EndpointType.Primary,
+                               string version = "1.0",
+                               string name = "") : this(type, name, version)
+        {
```

Comment:
The parameter order should be consistent in different constructors. (type, name, version)

Reply:
name = string.Empty

Reply:
`string.Empty` is not a compile-time constant.